mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -589,6 +602,9 @@ class ALS(@Since("1.4.0") override val uid: String) extends Estimator[ALSModel]
   @Since("2.2.0")
   def setColdStartStrategy(value: String): this.type = set(coldStartStrategy, value)
 
+  @Since("2.3.0")
+  def setThreshold(value: Int): this.type = set(threshold, value)
```

Comment:
I'm not sure callers can meaningfully understand and set this. Can't we pick a threshold programmatically?

Reply:
Yes, my test results is better than the previous results,  especially for native BLAS.  I will update my test results here soon, and I will change this set.